### PR TITLE
Add Initial Maven Project with Asciidoctor Plugin

### PR DESCRIPTION
**Pull request by jasoet-gdp**
Maven Project only generate PDF for now. Will add other `backend` later.

Source Docs Location = `src/main/asciidoc`
Destination Docs Location = `target/generated-docs`

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        <plugins>
+            <plugin>
+                <groupId>org.asciidoctor</groupId>
+                <artifactId>asciidoctor-maven-plugin</artifactId>
+                <version>${asciidoctor.maven.plugin.version}</version>
+                <dependencies>
+                    <dependency>
+                        <groupId>org.asciidoctor</groupId>
+                        <artifactId>asciidoctorj-pdf</artifactId>
+                        <version>1.5.0-alpha.9</version>
```
Comment (danieldietrich) [inline pom.xml]: I would move the to the properties version. Then we are able to update all versions to the newest version via

```java
mvn versions:update-properties
```

However, we need to add the versions plugin - I will create a separate issue for that.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                        <artifactId>jruby-complete</artifactId>
+                        <version>${jruby.version}</version>
+                    </dependency>
+                    <!-- Uncomment the following element to override the version of AsciidoctorJ -->
+                    <!--
+                    <dependency>
+                        <groupId>org.asciidoctor</groupId>
+                        <artifactId>asciidoctorj</artifactId>
+                        <version>1.5.3-SNAPSHOT</version>
+                    </dependency>
```
Comment (danieldietrich) [inline pom.xml]: What's the use-case for overriding the version? We may change the version locally in the properties section.

File: pom.xml (hunk 1 above)
Comment (jasoet-gdp) [inline pom.xml]: Noted. I'll fix it immediately

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.javaslang</groupId>
+    <artifactId>javaslang-docs</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
```
Comment (danieldietrich) [inline pom.xml]: Any ideas about versioning?

Suggestion: We could sync the version number with the code `javaslang` project, i.e. `2.0.0-SNAPSHOT`.

On the other hand the docs project will probably not be released for every minor `javaslang` version. But we should release the docs at least for every major release.

Comment (jasoet-gdp) [inline pom.xml]: I agree if we sync version number with `javaslang` project. 
How about if docs grow faster/slower than `javaslang` project i.e. when `javaslang` 2.0.0 released but docs only cover 50% of the API?

File: pom.xml (hunk 2 above)
Comment (jasoet-gdp) [inline pom.xml]: I decided to remove that `dependency` since `asciidoctorj` will be picked by `asciidoctorj-pdf`, any suggestion?

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                              - don't set linkcss attribute
+                              - set imagesdir attribute to path relative to AsciiDoc source file
+
+                              <attributes>
+                                  <imagesdir>./images</imagesdir>
+                              </attributes>
+
+                            IMPORTANT: When you enable image embedding, you must qualify the path the the imagesdir, as shown above.
+                            -->
+                            <attributes>
```
Comment (jasoet-gdp) [inline pom.xml]: Do not know best practice for generating `html5` using `asciidoctor` yet. Better leave comment for future reference.

Comment (danieldietrich) [inline pom.xml]: Yes, this is good. Thx for your investigation - we will check the best practices, e.g. by looking at other projects.

**Comment by danieldietrich** [comment]
This is a great initial commit! I will merge it now and address further todos in separate issues.

Many thanks!

File: pom.xml (hunk 3 above)
Comment (danieldietrich) [inline pom.xml]: Here is a follow-up regarding versioning: #3

File: pom.xml (hunk 4 above)
Comment (danieldietrich) [inline pom.xml]: Here is a follow-up regarding html5 generation with asciidoctor: #4

File: pom.xml (hunk 2 above)
Comment (danieldietrich) [inline pom.xml]: Please see #5.

**Pull request merged**